classes/data-stores/ActionScheduler_DBStore.php

```diff
@@ -37,21 +37,47 @@ public function init() {
 	}
 
 	/**
-	 * Save an action.
+	 * Save an action, checks if this is a unique action before actually saving.
+	 *
+	 * @param ActionScheduler_Action $action         Action object.
+	 * @param \DateTime              $scheduled_date Optional schedule date. Default null.
+	 *
+	 * @return int                  Action ID.
+	 * @throws RuntimeException     Throws exception when saving the action fails.
+	 */
+	public function save_unique_action( ActionScheduler_Action $action, \DateTime $scheduled_date = null ) {
+		return $this->save_action_to_db( $action, $scheduled_date, true );
+	}
+
+	/**
+	 * Save an action. Can save duplicate action as well, prefer using `save_unique_action` instead.
 	 *
 	 * @param ActionScheduler_Action $action Action object.
-	 * @param DateTime              $date Optional schedule date. Default null.
+	 * @param \DateTime              $scheduled_date Optional schedule date. Default null.
 	 *
 	 * @return int Action ID.
 	 * @throws RuntimeException     Throws exception when saving the action fails.
 	 */
-	public function save_action( ActionScheduler_Action $action, \DateTime $date = null ) {
-		try {
+	public function save_action( ActionScheduler_Action $action, \DateTime $scheduled_date = null ) {
```

Comment:
I don't think any action is needed here—it makes sense because the second param is `DateTime $scheduled_date = NULL` in the base class—but I wonder if it could cause issues if a user was interacting with the concrete object using named params in their method calls?

Probably safe to ignore, though (because it should align with the interface provided by the abstract class, and because we encourage use of top-level functions and not methods like this one).

Reply:
Yeah, I think we should be able to ignore this since having consistency with the base class is probably better.